tests/browser/test/basic_test.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Node.js script to run Automated tests in Chrome, via webdriver.
+ */
+/* eslint-env node */
+/* eslint-env mocha */
+/* eslint-env webdriverio */
+const webdriverio = require('webdriverio');
+const {posixPath} = require('../../../scripts/helpers');
+const chai = require('chai');
+const path = require("path");
+
+let browser;
+suite("Testing Basic Browser Test", function(done) {
+  this.timeout(0);
+  // Setup Selenium for each of the test
+  suiteSetup(async function() {
+    const options = {
+      capabilities: {
+        "browserName": 'chrome',
+        'goog:chromeOptions': {
+          args: ['--allow-file-access-from-files'],
+        },
+      },
+      services: [
+        ['selenium-standalone'],
+      ],
+      logLevel: 'warn',
+    };
+  
+    // Run in headless mode on Github Actions.
+    if (process.env.CI) {
+      options.capabilities['goog:chromeOptions'].args.push(
+          '--headless', '--no-sandbox', '--disable-dev-shm-usage',);
+    } else {
+      // --disable-gpu is needed to prevent Chrome from hanging on Linux with
+      // NVIDIA drivers older than v295.20. See
+      // https://github.com/google/blockly/issues/5345 for details.
+      options.capabilities['goog:chromeOptions'].args.push('--disable-gpu');
+    }
+  // Use Selenium to bring up the page
+    const url = 'file://' + posixPath(path.join(__dirname, '..')) + '/index.html';
+    console.log('Starting webdriverio...');
+    browser = await webdriverio.remote(options);
+    console.log('Loading URL: ' + url);
+    await browser.url(url);
+    return browser;
+  });
+
+
+test('Drag Block Event', async function() {
+  const block = await browser.$('[data-id="test_block"]');
```

Comment:
Love being able to query based on the data-id 🗡️ 

I think we should consider making the IDs of blocks in flyouts consistent with the IDs of blocks they create in the workspace (e.g. generating a new flyout block with a new ID every time one is "used" to create a new block in the workspace), so that we can also use the data-id in flyout tests.

This would be a change in core though, and idk if it would work with block recycling.

Reply:
That would be super helpful from a testing perspective!

Reply:
@maribethb do you have an opinion / context on this from working on the continuous toolbox plugin?

Reply:
I don't think I was the one that added block recycling logic to continuous toolbox so I don't really have any additional context there. I might worry it would break tests where it's assumed the "same" block in the flyout always has the same id.